tests/test_config_parser.py

```diff
@@ -124,6 +126,30 @@ def __call__(self, a, b):
     1,
     [0, 4],
 ]
+TEST_CASE_WRAPPER_ENABLED = [
+    {
+        "dataset": {
+            "_target_": "Dataset",
+            "data": [1, 2],
+            CONFIG_COMPONENT_KEY_WRAPPER: {"_target_": "CacheDataset", "_mode_": "callable"},
+        }
+    },
+    ["dataset", f"dataset#{CONFIG_COMPONENT_KEY_WRAPPER}"],
+    [CacheDataset, type(CacheDataset)],
```

Comment:
```suggestion
    [CacheDataset, CacheDataset],
```
The expression `type(CacheDataset)` returns `type` which allows the test to pass but isn't specific enough to be testing what we want, right?

Reply:
The problem is that this is not a instance of this class, since it has mode: callable and therefore will just be the class. And reusing the existing test setups, this is the the workaround that I found. I agree it's not very specific, and if you think it makes sense I can introduce some custom test wrapper class instead of trying to use some pre-existing classes from another library.

Reply:
I think the test might have to instantiate things further, checking that something is an instance of `type` is going to be true for any class.

Reply:
I now added a helper class that makes a more specific assert possible